tibber/tibber_home.py

```diff
@@ -429,16 +429,20 @@ async def _start():
                 except Exception:  # pylint: disable=broad-except
                     if self.rt_subscription_running:
                         _LOGGER.exception("Error in rt_subscribe")
-                    await asyncio.sleep(10)
-                    await asyncio.gather(
-                        *[
-                            self.update_info(),
-                            self._tibber_control.update_info(),
-                        ]
-                    )
+                    try:
+                        await asyncio.sleep(10)
+                        await asyncio.gather(
+                            *[
+                                self.update_info(),
+                                self._tibber_control.update_info(),
+                            ]
+                        )
+                    except Exception:
+                        await asyncio.sleep(10)
+
                     if not self.has_real_time_consumption:
                         _LOGGER.error("No real time device for %s", self.home_id)
-                        return
+                        continue
```

Comment:
If there are no real-time device anymore, we should return

Reply:
Ok. What I see when the code is running is that following a network connection issue, the connection is reestablished, but there is no _start() task running to actually read the data that is available.... And i do not see that a new one is started from anywhere.

I have tested the above code by running the README example, and it is reconnecting and continuing to receive data. 

If the _start() task is stopped, then you need to start a new one to handle the data...

Reply:
This is a log from running the version in the pull request... I'm disconnecting from the wifi, and after a short time, I reconnect, and data reception is picked up again.

The message "No real time device for xxx" appears, and would have terminated the _start() task if implemented with return statement.

{'accumulatedConsumption': 41.442, 'accumulatedConsumptionLastHour': 1.091, 'accumulatedCost': 28.11327, 'accumulatedProduction': 0.004, 'accumulatedProductionLastHour': 0, 'accumulatedReward': 0.003896, 'averagePower': 2012.5, 'currency': 'SEK', 'currentL1': 2.7, 'currentL2': 4.1, 'currentL3': 3.3, 'lastMeterConsumption': 43385.802, 'lastMeterProduction': 6436.58, 'maxPower': 7256, 'minPower': 0, 'power': 1739, 
'powerFactor': 0.875, 'powerProduction': 0, 'powerReactive': 0, 'signalStrength': None, 'timestamp': '2023-02-17T20:35:24.000+01:00', 'voltagePhase1': 228, 'voltagePhase2': 227.7, 'voltagePhase3': 229.4, 'estimatedHourConsumption': 1.887}
Error in rt_subscribe
Traceback (most recent call last):
  File "c:\Users\Knut-Lenovo-E580\Python\pyTibberDaniel\pyTibber\tibber\tibber_home.py", line 415, in _start
    async for data in self._tibber_control.sub_manager.session.subscribe(
  File "C:\Users\Knut-Lenovo-E580\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\gql\client.py", line 1064, in subscribe
    async for result in inner_generator:
  File "C:\Users\Knut-Lenovo-E580\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\gql\client.py", line 960, in _subscribe
    async for result in inner_generator:
  File "C:\Users\Knut-Lenovo-E580\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\gql\transport\websockets_base.py", line 388, in subscribe
    answer_type, execution_result = await listener.get()
  File "C:\Users\Knut-Lenovo-E580\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\gql\transport\websockets_base.py", line 56, in get    raise item
gql.transport.exceptions.TransportServerError: No pong received after 5.0 seconds
Watchdog: Connection is down
No real time device for 96a14971-525a-4420-aae9-e5aedaa129ff
No real time device for 96a14971-525a-4420-aae9-e5aedaa129ff
No real time device for 96a14971-525a-4420-aae9-e5aedaa129ff
No real time device for 96a14971-525a-4420-aae9-e5aedaa129ff
{'data': {'liveMeasurement': {'accumulatedConsumption': 41.486, 'accumulatedConsumptionLastHour': 1.135, 'accumulatedCost': 28.139636, 'accumulatedProduction': 0.004, 'accumulatedProductionLastHour': 0, 'accumulatedReward': 0.003896, 'averagePower': 2012.2, 'currency': 'SEK', 'currentL1': 2.8, 'currentL2': 4.1, 'currentL3': 3.3, 'lastMeterConsumption': 43385.846, 'lastMeterProduction': 6436.58, 'maxPower': 7256, 
'minPower': 0, 'power': 1775, 'powerFactor': 0.878, 'powerProduction': 0, 'powerReactive': 0, 'signalStrength': None, 'timestamp': '2023-02-17T20:36:53.000+01:00', 'voltagePhase1': 227.5, 'voltagePhase2': 227.3, 'voltagePhase3': 228.9, 'estimatedHourConsumption': 1.887}}}
{'accumulatedConsumption': 41.486, 'accumulatedConsumptionLastHour': 1.135, 'accumulatedCost': 28.139636, 'accumulatedProduction': 0.004, 'accumulatedProductionLastHour': 0, 'accumulatedReward': 0.003896, 'averagePower': 2012.2, 'currency': 'SEK', 'currentL1': 2.8, 'currentL2': 4.1, 'currentL3': 3.3, 'lastMeterConsumption': 43385.846, 'lastMeterProduction': 6436.58, 'maxPower': 7256, 'minPower': 0, 'power': 1775, 'powerFactor': 0.878, 'powerProduction': 0, 'powerReactive': 0, 'signalStrength': None, 'timestamp': '2023-02-17T20:36:53.000+01:00', 'voltagePhase1': 227.5, 'voltagePhase2': 227.3, 'voltagePhase3': 228.9, 'estimatedHourConsumption': 1.887}

Reply:
Ok, I will look at it. This is not how it should be fixed